lib/Parse/ParsePattern.cpp

```diff
@@ -787,6 +788,10 @@ Parser::parseFunctionSignature(Identifier SimpleName,
     } else if (Tok.is(tok::kw_rethrows)) {
       throwsLoc = consumeToken();
       rethrows = true;
+    } else if (Tok.is(tok::kw_throw)) {
+        throwsLoc = consumeToken();
+        isThrow = true;
+        return diagnose(throwsLoc, diag::throw_in_function_type_and_wrong_position);
```

Comment:
Why not put all of this [here](https://github.com/apple/swift/blob/592d6603afe6646db59af57fb7cc39e21d679f21/lib/Parse/ParsePattern.cpp#L786)?

Reply:
Also, you need to indent your code with 2 spaces. You can run `git-clang-format` for this - you can run this when you're committing the file.

Reply:
Sorry I don't understand the question. The linked `if` is checking for `throws` but the new code needs to check for `throw`.

Reply:
I meant you could combine it i.e “if token == throw OR token == throws” and then consume the token and set isThrow to true.

Then, check if you set isThrow=true and return the diagnostic. 

Anyway, it was just an optional suggestion :)